wikitextprocessor/core.py

```diff
@@ -662,12 +662,12 @@ def repl_extlink(m):
                     break
             # Replace template invocation
             text = re.sub(
-                r"{" + MAGIC_NOWIKI_CHAR + r"?{(("
+                r"{" + MAGIC_NOWIKI_CHAR + r"?{((?:"
                 r"{\|[^{}]*?\|}|"  # Outer table tokens
                 r"}[^{}]|" # lone }????
                 r"[^{}](?:{[^{}|])?|"  # lone possible {???
                 r"-{}-|"  # GitHub issue #59 Chinese wiktionary special `-{}-`
-                r"[^{}]+{[^{}]+}"  # GitHub issue #72 zh.wiktionary `A{pron}`
+                r"[^{}]{[^{}]+}"  # GitHub issue #72 zh.wiktionary `A{pron}`
```

Comment:
This might be triggered by `A{aa}}`, so might need to add something like `[^}]` at the end, but then it won't catch `{{A{aa}}}`...

Reply:
Maybe the regex could be simplified to `{{((?:[^{}]|-{}-|{[^{}]+})+?)}}` but I haven't test it on the dump file, all the unit tests are passed.

Reply:
Tatu arrived an hour ago and we've been looking at the regex, I'll be simplifying some of the regex, but he was puzzled that the regex from before issue #72 didn't catch the curly braces. That's what the two mystery lines are for:

```
                r"}[^{}]|" # lone }????
                r"[^{}](?:{[^{}|])?|"  # lone possible {???
```
These should catch any lone curly braces, which the ones in `a{aa}` should be. I'll test this out with a regex tester.

Reply:
On https://regex101.com/ (switch to the Python engine), `{{(({\|[^{}]*?\|}|}[^{}]|[^{}](?:{[^{}|])?)+?)}}` is able to pick up `{{foo a{a} }}`; does this work with zh-pages? I left out the "-{}-" because it wasn't relevant, but it needs to be part of the regex as a special case.

Reply:
Tatu also wished for the curly braces to be escaped, just in case.

Reply:
> On https://regex101.com/ (switch to the Python engine), `{{(({\|[^{}]*?\|}|}[^{}]|[^{}](?:{[^{}|])?)+?)}}` is able to pick up `{{foo a{a} }}`; does this work with zh-pages? I left out the "-{}-" because it wasn't relevant, but it needs to be part of the regex as a special case.

This doesn't work with `{{zh-x|約 有 6 '''%'''{pā} 的 臺灣人 血型 是 A{ēi}B{bī}型。|}}`! Weird!

Reply:
It works if you remove parts of the `{bī}`. An issue with regex consumption.

Reply:
The issue was with B ovelapping in the above case, which could be fixed with a regex lookahead (had to look that up, and yep, modern Python has regex lookahead): `{{(({\|[^{}]*?\|}|}(?=[^{}])|[^{}]{[^{}|]|[^}{])+?)}}`, `(?=...)` look for `...` ahead, but do not consume. It works with the example from above.